File: RuntimeData/xnaruntime/RuntimeXNA/RuntimeXNA/Expressions/EXP_ROUND.cs
//----------------------------------------------------------------------------------
//
// ROUND
//
//----------------------------------------------------------------------------------
using System;
using RuntimeXNA.RunLoop;
namespace RuntimeXNA.Expressions
{
	
	public class EXP_ROUND:CExp
	{
		
		public override void  evaluate(CRun rhPtr)
		{
			rhPtr.rh4CurToken++;
			double value = rhPtr.get_ExpressionDouble();
			double v = System.Math.Floor(value);
			if (value - v > 0.5)
				v++;
            rhPtr.getCurrentResult().forceInt((int)v);
		}
	}
}

File: RuntimeData/runtime/app/src/main/java/Expressions/EXP_ROUND.java
/* Copyright (c) 1996-2013 Clickteam
 *
 * This source code is part of the Android exporter for Clickteam Multimedia Fusion 2.
 * 
 * Permission is hereby granted to any person obtaining a legal copy 
 * of Clickteam Multimedia Fusion 2 to use or modify this source code for 
 * debugging, optimizing, or customizing applications created with 
 * Clickteam Multimedia Fusion 2.  Any other use of this source code is prohibited.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
 * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
 * IN THE SOFTWARE.
 */
//----------------------------------------------------------------------------------
//
// ROUND
//
//----------------------------------------------------------------------------------
package Expressions;

import RunLoop.*;

public class EXP_ROUND extends CExp
{
    
    public void evaluate(CRun rhPtr)
    {        
	rhPtr.rh4CurToken++;
	double value=rhPtr.get_ExpressionDouble();
	double v=Math.floor(value);
	if (value-v>0.5)
	    v++;
	rhPtr.rh4Results[rhPtr.rh4PosPile].forceInt((int)v);
    }
    
}
